dataRetriever/dataPool/proofsCache/proofsPool.go

```diff
@@ -0,0 +1,106 @@
+package proofscache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/pkg/errors"
+)
+
+// ErrMissingProof signals that the proof is missing
+var ErrMissingProof = errors.New("missing proof")
+
+// ErrNilProof signals that a nil proof has been provided
+var ErrNilProof = errors.New("nil proof provided")
```

Comment:
add them to a new file, errors.go or dataRetriever/errors.go

Reply:
done